ubuntu22.04/nvidia-driver

```diff
@@ -481,6 +481,31 @@ _shutdown() {
     return 1
 }
 
+_resolve_kernel_type() {
+  # For backwards compatibility with older GPU Operator versions where KERNEL_MODULE_TYPE is not set,
+  # honor the deprecated OPEN_KERNEL_MODULES_ENABLED field
+  if [ -z "${KERNEL_MODULE_TYPE}" ]; then
+    [[ "${OPEN_KERNEL_MODULES_ENABLED}" == "true" ]] && KERNEL_TYPE=kernel-open || KERNEL_TYPE=kernel
+    return 0
+  fi
+
+  if [ "${KERNEL_MODULE_TYPE}" == "proprietary" ]; then
+    KERNEL_TYPE=kernel
+  elif [ "${KERNEL_MODULE_TYPE}" == "open" ]; then
+    KERNEL_TYPE=kernel-open
+  elif [ "${KERNEL_MODULE_TYPE}" == "auto" ]; then
+      kernel_module_type=$(nvidia-installer --print-recommended-kernel-module-type)
```

Comment:
Shall we move the `nvidia-installer` to the `PATH` during build-time?

Reply:
Possibly. That would require us to extract the runfile during build time, which we aren't doing currently.

Reply:
Fair enough; we probably shouldn't extract it during build-time, but we will still need to move the `nvidia-installer` to the PATH

Reply:
Yep. Extracting at build time will increase our image size significantly. We will need to move it to the path or `cd` into the local directory before invoking `_resolve_kernel_type` (like we do elsewhere).

Reply:
or cd into `NVIDIA-Linux-$DRIVER_ARCH-$DRIVER_VERSION`

Reply:
That assumes the runfile has been extracted beforehand though.

Reply:
> Yep. Extracting at build time will increase our image size significantly. We will need to move it to the path or cd into the local directory before invoking _resolve_kernel_type (like we do elsewhere).

Ah, just seeing this comment now. Sounds good

Reply:
I pushed an update, which uses the absolute path to the `nvidia-installer`. It assumes the runfile has been extracted prior to the call to `_resolve_kernel_type`.